clustering_bounds_comparator.hh

```diff
@@ -149,6 +149,10 @@ public:
         bool inclusive = bv._kind != bound_kind::excl_end && bv._kind != bound_kind::excl_start;
         return {typename R<clustering_key_prefix_view>::bound(bv._prefix.get().view(), inclusive)};
     }
+    static bool ranges_overlap(const schema& s, bound_view s1, bound_view e1, bound_view s2, bound_view e2) {
```

Comment:
There's `position_range::overlaps()` which you could reuse. We should move away from bound_view.

I didn't check if this implementation is correct.